adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepository.java

```diff
@@ -482,4 +487,45 @@ public FareRule getFareRule(final String fareId, final String routeId, final Str
         return fareRuleCollection.get(FareRule.getFareRuleMappingKey(fareId, routeId, originId, destinationId,
                 containsId));
     }
+
+    /**
+     * Add a shape point to a shape. A shape is a list of shape points whereas a ShapePoint represents a row from
+     * shapes.txt to this. Return the entity added to the repository if not null, otherwise returns null.
```

Comment:
I would re-word this:

>Return the entity added to the repository if not null, otherwise returns null.

to better describe the exact behavior:

>Return the entity added to the repository if the entity was successfully added, and returns null if the provided newShapePoint already exists in the repository.